### PR TITLE
Introduce 64-bit position type for storing positions in the game world.

### PR DESCRIPTION
**Pull request by dgavedissian**
TLDR is that all spatial positions in the engine use the new `Position` type (which is a double vector). Then, the renderer, audio and physics modules are adapted to use the new Position types.

Note that this is only part 1, this doesn't actually solve large world support YET. This simply introduces the new Position type and wired it up in all the places I could think of.

Renderer will render everything relative to the camera. That means that when we compute the model-view-projection matrix, the model matrix uses the f32 offset as its translation component, and the view matrix has (0,0,0) as its translation component. Any positions passed to shaders as uniforms also need to be translated in a similar fashion. For now, we just treat "(0,0,0)" as the camera position before we wire things up.

Physics already is based on f64 positions, so we just need to convert them.

Audio uses a "floating origin" setup, where we recompute the position of all emitters and listeners when the camera moves far enough away from the last updated origin.

### CODE REVIEW

File: engine/lib/luajit-ffi-gen/src/impl_item/type_info.rs [hunk 1]
```diff
@@ -29,6 +29,7 @@ const COPY_TYPES: &[&str] = &[
     "DVec4",
     "Vec4",
     "Box3",
+    "Position",
```
Comment (HaronK) [inline engine/lib/luajit-ffi-gen/src/impl_item/type_info.rs:32]: This list is for types that are copyable in LuaJit terms and Position type (as well as other Vec3/4 types) cannot be propery copied by Lua.
We should remove all Vec3/4 and Box3 types from this list.
I have this as a separate task but if you already working on it, can you probably do this cleanup? Unless it causes a lot of refactoring.
At least Position should not be here.

Comment (dgavedissian) [reply engine/lib/luajit-ffi-gen/src/impl_item/type_info.rs:35]: I'm going to leave it for now, and deal with this separately. When removing Position here, it seems to cause other problems in luajit-ffi-gen which I can't recall at this moment.

File: script/GameObjects/Elements/Material/RigidBody.lua [hunk 2]
```diff
@@ -191,12 +191,12 @@ end
...
 function Entity:getToLocalMatrix()
     assert(self.body)
-    return self.body:getToLocalMatrix()
+    return self.body:getToLocalMatrix(Position.Identity())
 end
 
 function Entity:getToWorldMatrix()
     assert(self.body)
-    return self.body:getToWorldMatrix()
+    return self.body:getToWorldMatrix(Position.Identity())
```
Comment (dgavedissian) [inline script/GameObjects/Elements/Material/RigidBody.lua:199]: Here's the real meat. For now this doesn't do anything (the 64-bit positions are just converted to 32-bit ones), but in a future PR I will be passing the camera position here, and adjusting the view matrix to compensate for it.

**Review by HaronK — APPROVED** [review]
Overall looks good. I see there are multiple places where coordinates are converted to and from Position. I hope it's either proper solution or temporary fix. Otherwise we could loose precision.

**Comment by dgavedissian** [comment]
@HaronK All of those are for calculating the initial placement of planets and things, where losing precision doesn't matter that much. I do plan to adjust those slightly in the next PR, for example:

```
planet:setPos(rng:getDir3():scale(Config.gen.scaleSystem * (1.0 + rng:getExp())):toPosition())
```
will become
```
planet:setPos(rng:getDir3():toPosition():scale(Config.gen.scaleSystem * (1.0 + rng:getExp())))
```

File: engine/lib/phx/src/physics/rigid_body.rs [hunk 3]
```diff
@@ -764,9 +768,7 @@ impl RigidBody {
         let my_position = self.get_position();
         let target_position = target.get_position();
 
-        let distance = my_position.distance(target_position);
-
-        distance
+        my_position.distance(target_position) as f32
```
[the remark this answers is not in the document]
Comment (dgavedissian) [reply engine/lib/phx/src/physics/rigid_body.rs:771]: Yeah probably, I'll address this in https://github.com/Limit-Theory-Redux/ltheory/pull/281
